src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -495,17 +586,18 @@ function ProtectEnvironmentalVariables(): void {
   process.env.GOOGLE_APPLICATION_CREDENTIALS = "";
 }
 
-function InitializeEnvironmentalVariables(projectId: string): void {
+function InitializeEnvironmentalVariables(frb: FunctionsRuntimeBundle): void {
+  const projectId = frb.projectId;
+
   process.env.GCLOUD_PROJECT = projectId;
   process.env.FUNCTIONS_EMULATOR = "true";
-  /*
-    Do our best to provide reasonable FIREBASE_CONFIG, based on firebase-functions implementation
-    https://github.com/firebase/firebase-functions/blob/master/src/index.ts#L70
-   */
+
+  // Do our best to provide reasonable FIREBASE_CONFIG, based on firebase-functions implementation
+  // https://github.com/firebase/firebase-functions/blob/59d6a7e056a7244e700dc7b6a180e25b38b647fd/src/setup.ts#L45
   process.env.FIREBASE_CONFIG = JSON.stringify({
     databaseURL: process.env.DATABASE_URL || `https://${process.env.GCLOUD_PROJECT}.firebaseio.com`,
     storageBucket: process.env.STORAGE_BUCKET_URL || `${process.env.GCLOUD_PROJECT}.appspot.com`,
-    projectId: process.env.GCLOUD_PROJECT,
+    projectId: projectId || process.env.GCLOUD_PROJECT,
```

Comment:
Is this redundant with line 592, since projectId and GCLOUD_PROJECT should be equal ?